hetmech/diffusion.py

```diff
@@ -37,18 +37,21 @@ def diffusion_step(
     numpy.ndarray
         Normalized matrix with dtype.float64.
     """
-    # returns a newly allocated numpy.ndarray
-    matrix = numpy.array(matrix, numpy.float64, copy=copy)
+    # returns a newly allocated array
+    mat_type = type(matrix)
```

Comment:
These few lines appear in multiple places... should they be pulled out into a function? We want to minimize the type logic in these functions for the sake of readability.